core/src/main/java/io/grpc/ServerCall.java

```diff
@@ -180,6 +180,9 @@ public void setMessageCompression(boolean enabled) {
    * The compressor to use will be looked up in the {@link CompressorRegistry}.  Default gRPC
    * servers support the "gzip" compressor.
    *
+   * <p>It is safe to call this even if the client does not support the compression format chosen.
```

Comment:
Optional: you might call out that the analogous client side call (CallOptions.withCompression) does not safely negotiate.